### DISCUSSION TITLE
A symbolic used as a index of numpy array and introduced into the cost function

### DISCUSSION THREAD

**Discussion by AnjiLin12321** [casadi/casadi#3192] [Q&A]
Hello!
I would like to know if there is any way to use MX symbolics as the index of the array and introduce this information into the cost function to solve correctly
```
i=MX.sym('i')
A=np.array([3,5,1])  #The content of A comes from other places
obj=A[i]
```
When I tried to minimize 'obj' with a casadi solver, I found that 'i' would solve to the initial value. I would like to ask if my calling method is incorrect. Or is it because the solver needs to use gradients for calculation, which makes it impossible to calculate.

Thanks in advance!

**Comment by jgillis** [comment]
Syntactically, this should work, but indeed a lack of gradient information makes it hard to do anything useful with this formulation.
Perhaps you can use 'i' to perform interpolation? (we have linear or bspline)
